src/main/pages/setup-kubernetes/kubernetes-admin-guide.md

```diff
@@ -0,0 +1,208 @@
+
+* If you passed `CHE_INFRA_KUBERNETES_USERNAME` and `CHE_INFRA_KUBERNETES_PASSWORD` to Che deployment, those credentials will be used to create workspace objects.
+
+* If there's `CHE_INFRA_KUBERNETES_OAUTH__TOKEN` fabric8 client will use it. If both are available, token has a priority over username/pass.
+
+* If none are available **che service account** is used to create workspace objects. This is default behavior. Since che SA cannot create objects outside a namespace it's bound to, all workspace objects are created in Che namespace. Admins can grant `che` [service account super user permissions](https://kubernetes.io/docs/admin/authorization/rbac/#service-account-permissions), and this way it will be possible to use this SA to create objects outside Che namespace. OpenShift cluster admins can do it this way using oc CLI:
+
+```bash
+oc adm policy add-cluster-role-to-user self-provisioner system:serviceaccount:eclipse-che:che
+
+# eclipse-che is Che namespace
+```
+
+Advantages and disadvantages of using token, username/pass, and service account:
+
+|Auth type         | Pros                                         | Cons |
+|**Token**             | Objects can be created outside Che namespace | Tokens tend to expire
+|**Username/Password** | Objects can be created outside Che namespace | If username/pass are changed, they need to be updated in Che deployment as well. Many authentication types like oAuth make it impossible to login with username/pass
+| **Service account**   | No expiration risks                          | Cannot create objects outside its namespace unless given admin privileges
+
+## Storage Overview
+
+Che server, Keycloak and Postgres pods, as well as workspace pods use PVCs that are then bound to PVs with [ReadWriteOnce access mode](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#access-modes). Che PVCs are defined in deployment yamls, while [workspace PVC](#che-workspaces-storage) access mode and claim size is configurable through Che deployment environment variables.
+
+## Che Infrastructure: Storage
+
+* Che server claims 1GB to store logs and initial workspace stacks. PVC mode is RWO
+* Keycloak needs 2 PVCs, 1Gb each to store logs and Keycloak data
+* Postgres needs one 1GB PVC to store db
+
+## Che Workspaces: Storage
+
+As said above, che workspace PVC access type and claim size is configurable, and so is a workspace PVC strategy:
+
+|strategy       | details       | pros | cons |
+|**common**|One PVC for all workspaces, sub-paths pre-created| easy to manage and control storage. no need to recycle PVs when pod with pvc is deleted | ws pods should all be in one namespace
+|**unique**|PVC per workspace| Storage isolation | An  undefined number of PVs is required |
+
+## Common PVC strategy
```

Comment:
Headings should consistently use title case.

Reply:
fixed